core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/edit-ema-palette/edit-ema-palette.component.ts

```diff
@@ -0,0 +1,183 @@
+import { Subject } from 'rxjs';
+
+import { AsyncPipe, NgIf } from '@angular/common';
+import {
+    CUSTOM_ELEMENTS_SCHEMA,
+    ChangeDetectionStrategy,
+    Component,
+    EventEmitter,
+    Input,
+    OnDestroy,
+    OnInit,
+    Output,
+    inject
+} from '@angular/core';
+import { FormControl } from '@angular/forms';
+
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+
+import { DotESContentService, DotPropertiesService } from '@dotcms/data-access';
+import { DotContainerStructure, DotPageContainerStructure } from '@dotcms/dotcms-models';
+
+import { EditEmaPaletteContentTypeComponent } from './components/edit-ema-palette-content-type/edit-ema-palette-content-type.component';
+import { EditEmaPaletteContentletsComponent } from './components/edit-ema-palette-contentlets/edit-ema-palette-contentlets.component';
+import { PALETTE_TYPES } from './shared/edit-ema-palette.enums';
+import { DotPaletteStore } from './store/edit-ema-palette.store';
+
+@Component({
+    selector: 'dot-edit-ema-palette',
+    standalone: true,
+    templateUrl: './edit-ema-palette.component.html',
+    styleUrls: ['./edit-ema-palette.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    imports: [
+        NgIf,
+        AsyncPipe,
+        EditEmaPaletteContentTypeComponent,
+        EditEmaPaletteContentletsComponent
+    ],
+    providers: [DotPaletteStore, DotESContentService],
+    schemas: [CUSTOM_ELEMENTS_SCHEMA]
+})
+export class EditEmaPaletteComponent implements OnInit, OnDestroy {
+    @Input() languageId: number;
+    @Input() containers: DotPageContainerStructure;
+
+    @Output() dragStart = new EventEmitter();
+    @Output() dragEnd = new EventEmitter();
+
+    private readonly store = inject(DotPaletteStore);
+    private readonly dotConfigurationService = inject(DotPropertiesService);
+    private destroy$ = new Subject<void>();
+
+    vm$ = this.store.vm$;
+    searchContenttype = new FormControl('');
+    searchContentlet = new FormControl('');
+
+    allowedContentTypes: string[];
+    currentContenttype = ''; //Search how avoid to use this, its neede in line 148.
```

Comment:
I don't understand why you need this. If you want we can speak about this and find a workaround